src/redeem.ts

```diff
@@ -143,6 +144,17 @@ export class Redeem {
             console.log("Parachain not connected");
             return;
         }
+        console.log(`[${new Date().toLocaleString()}] -----Cancelling expired redeems-----`);
+        const botAccountId = this.polkaBtc.api.createType("AccountId", account.address);
+        this.polkaBtc.redeem.subscribeToRedeemExpiry(botAccountId, (redeemId: H256) => {
```

Comment:
I think a way to optimize this would be to do the callback and then calling a `cancelExpiredRedeem(redeemId: H256)` directly on the result. That way you don't have to add all the elements to an array.
As an alternative, you could fetch all the redeem requests for the users and filter for the expired ones. This would allow you to get rid of the magic sleep value that might be nondeterministic depending on response times from the parachain etc.

Reply:
I'll just remove the `sleep`. It doesn't really matter to be honest, it was just a convenience for me while testing. You can't call `cancelExpiredRedeem` inside the callback, because the callback is called by the polkadot-js subscription which doesn't `await`. A quick succession of un`await`ed callbacks will result in tx priority errors.

> you could fetch all the redeem requests for the users and filter for the expired ones

The callback from the lib does just this